Adding Registration functionality

kat-kime opened the pull request:
Close #3 
Close #2 

- Adds RegisterActivity.class
- Initializes Parse SDK
- Allows user to login
- Allows user to register

Allows user to successfully register a new account.

kat-kime committed:
Adding Login and Register Screens

Created .xml files for activity_login and activity_register.

kat-kime committed:
Updating String Resources

Adding string resources including:

- Login
- Sign Up
- Password
- Etc

kat-kime committed:
Adding Login Functionality

Added the following classes:

- ParseApplication
- LoginActivity

And made changes to the following files:

- AndroidManifest.xml
- build.gradle (root and app)

These allow users to log in to our Parse Database.

kat-kime committed:
Adding Register Functionality

Allows user to register successfully.

mlapresta commented on app/build.gradle:
```diff
@@ -1,10 +1,19 @@
 
     defaultConfig {
+        // should correspond to key/value pairs inside the file
+        buildConfigField("String", "CLIENT_KEY", apikeyProperties['CLIENT_KEY'])
```
It might be helpful to have a apikey.properties.example file with blank CLIENT_KEY / APP_ID fields, and have this in the project folder. Then when we add instruction in the readme, we can add to fill that in w/ credentials
We should probably add the apikey.properties file to our .gitignore file as well

kat-kime replied:
Ah, yes - this is where my limited view of GitHub comes in - I added it somewhere in a different pull request, eek!
Either way, I added it again to this PR - can you let me know if you see it?

mlapresta replied:
Perfect! No, you're totally good, I was able to see the added file. The last thing I would do is just rename it apikey.properties.example. 
If you keep it named apikey.properties, that means that if it gets added to the .gitignore, any changes won't be added to git (like if we were to place our Spoontacular API key in the same file. Also, whenever someone pulls down the PR, they'll have to switch back to another branch to grab their api keys if the file gets overwritten

kat-kime replied:
Changed to apikey.properties.example and added apikey.properties to gitignore

mlapresta commented on app/src/main/java/com/example/fond/LoginActivity.java:
```diff
@@ -0,0 +1,92 @@
+                // If e is not null, this means something went wrong
+                if (e != null) {
+                    Log.e(TAG, "Issue with login: " + e.toString());
+                    Toast.makeText(LoginActivity.this, "Login unsuccessful", Toast.LENGTH_SHORT).show();
```
Great job! I love the toast here to let the user know if there are issues :)

kat-kime committed:
Add Blank apikey.properties

Adding apikey.properties field with blank strings

kat-kime committed:
Changing apikey.properties to example

Changing apikey.properties to example so apikey.properties file isn't overwritten.

kat-kime committed:
Adding apikey.properties to gitignore

mlapresta reviewed (approved):
Yay! Great job!
